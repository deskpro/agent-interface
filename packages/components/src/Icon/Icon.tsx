import * as React from "react";
import "@deskpro/agent-interface-style/dist/elements/dp-Icons.css";
import icons from "./icons";

export type IconProps = {
  name: string;
  size?: number;
  color?: "default" | "primary" | "dark";
};

const colors = {
  default: "#979797",
  primary: "#3a8dde",
  dark: "#8B9293"
};

const Icon: React.SFC<IconProps> = ({ name, size = 24, color = "default" }) => {
  const SvgIcon = icons[name];
  if (!SvgIcon) {
    return null;
  }
  return (
    <span className={`dp-Icon dp-Icon--${name}`}>
      <SvgIcon
        fill={colors[color] || colors.default}
        width={size}
        height={size}
      />
    </span>
  );
};

Icon.defaultProps = {
  color: "default",
  size: 24
};

export default Icon;
