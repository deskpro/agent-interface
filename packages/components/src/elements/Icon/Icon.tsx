import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-Icons.css";
import icons from "./icons";

export type IconProps = React.ComponentProps<"span"> & {
  name: string;
  size?: number;
  circle?: boolean;
  color?: DPColor;
  iconRef?: React.Ref<HTMLSpanElement>;
  onClick?(
    e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>
  ): void;
};

const Icon: React.FC<IconProps> = ({
  name,
  size = 24,
  color,
  circle,
  iconRef,
  onClick,
  ...props
}) => {
  const SvgIcon = icons[name];
  if (!SvgIcon) {
    return null;
  }
  return (
    <span
      className={classNames(`dp-Icon Icon--${name}`, {
        "Icon--circle": circle,
        [`Icon--${color}`]: !!color
      })}
      role="button"
      tabIndex={-1}
      onClick={onClick}
      onKeyDown={onClick}
      ref={iconRef}
      {...props}
    >
      <SvgIcon width={size} height={size} />
    </span>
  );
};

Icon.defaultProps = {
  color: undefined,
  size: 24
};

export default React.memo(
  React.forwardRef<HTMLSpanElement, IconProps>((props, ref) => (
    <Icon {...props} iconRef={ref as React.Ref<HTMLSpanElement>} />
  ))
);
