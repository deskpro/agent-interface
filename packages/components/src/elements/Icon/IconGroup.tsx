import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/components/dp-ListPane.css";

export type IconGroupProps = {
  className?: string;
};

const IconGroup: React.SFC<IconGroupProps> = ({ children, className }) => (
  <span className={classNames("dp-IconGroup", className)}>{children}</span>
);

export default IconGroup;
