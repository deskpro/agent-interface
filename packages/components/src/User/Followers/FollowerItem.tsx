import * as React from "react";
import { Manager, Reference, Popper } from "react-popper";

import Icon from "../../elements/Icon/Icon";

export type FollowerItemProps = {
  id: React.Key;
  name: string;
  avatar?: React.ReactNode;
  onRemove: (itemId: React.Key) => void;
};

const FollowerItem: React.FC<FollowerItemProps> = ({
  id,
  name,
  avatar,
  onRemove
}) => {
  const [isHovered, setHoveredState] = React.useState(false);
  return (
    <Manager>
      <Reference>
        {({ ref }) => (
          <span
            ref={ref}
            className="dp-FollowersItem"
            onMouseEnter={() => setHoveredState(true)}
            onMouseLeave={() => setHoveredState(false)}
          >
            <span className="dp-LogoName">
              {avatar}
              <span className="dp-UserName">{name}</span>
            </span>
            <Icon name="close" onClick={() => onRemove(id)} size={8} />
          </span>
        )}
      </Reference>
      {isHovered && (
        <Popper
          positionFixed={false}
          placement="right-end"
          modifiers={{ offset: { offset: "20" } }}
        >
          {({ ref, style }) => (
            <span className="dp-ToolTips" ref={ref} style={style}>
              {name}
            </span>
          )}
        </Popper>
      )}
    </Manager>
  );
};

export default FollowerItem;
