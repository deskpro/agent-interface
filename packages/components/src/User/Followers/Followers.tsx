import * as React from "react";
import classNames from "classnames";
import Icon from "../../elements/Icon/Icon";
import Button from "../../Button/Button";

import "@deskpro/agent-interface-style/dist/components/dp-Sidebar.css";
import "@deskpro/agent-interface-style/dist/components/dp-Ticketview.css";
import FollowerItem from "./FollowerItem";

export interface FollowerInfo {
  id: React.Key;
  name: string;
  avatar?: React.ReactNode;
}

export type FollowersProps = {
  className?: string;
  style?: any;
  agents: FollowerInfo[];
  isFollowed?: boolean;
  onFollowToggle: () => void;
  onRemove: (userId: React.Key) => void;
  onAdd: () => void;
};

const Followers: React.FC<FollowersProps> = ({
  agents,
  className,
  isFollowed = false,
  onFollowToggle,
  onRemove,
  onAdd
}) => (
  <div className={classNames("dp-Followers", className)}>
    <div className="dp-SectionControls">
      <div className="dp-SectionTitle">
        <div className="dp-TitleControls">
          Followers <span className="dp-qt">({agents.length})</span>
        </div>
        <Button
          variant="link"
          onClick={onFollowToggle}
          className={classNames("dp-StatusTimes dp-FollowersStatus", {
            "is-active": isFollowed
          })}
        >
          {isFollowed ? "Following" : "Follow"}
          {isFollowed && <Icon name="following" color="primary" size={10} />}
        </Button>
      </div>
    </div>
    <div
      className={classNames("dp-FollowerList", {
        "Followers--Medium": agents.length > 7 && agents.length < 25,
        "Followers--Large": agents.length > 24
      })}
    >
      {agents.map(({ id, ...agentInfo }) => (
        <FollowerItem key={id} id={id} {...agentInfo} onRemove={onRemove} />
      ))}

      <span className="dp-FollowersBtn">
        <Button variant="link" size="small" onClick={onAdd}>
          <Icon name="add-follower" color="primary" size={10} />
          <span>Add</span>
        </Button>
      </span>
    </div>
  </div>
);

export default Followers;
