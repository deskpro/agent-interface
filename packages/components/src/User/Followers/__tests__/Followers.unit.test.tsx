import * as React from "react";
import { render, fireEvent } from "react-testing-library";

import Followers from "../Followers";
import Icon from "../../../elements/Icon/Icon";

describe("<Followers />", () => {
  it("should match the snapshot for single follower", () => {
    const { container } = render(
      <Followers
        agents={[
          {
            id: 1,
            name: "Artem Berdyshev",
            avatar: <Icon name="avatar" size={10} />
          }
        ]}
        onRemove={jest.fn()}
        onAdd={jest.fn()}
        onFollowToggle={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot("Single follower");
  });

  it("should match the snapshot for 10 followers", () => {
    const agents = Array.from({ length: 10 }, (_, idx) => ({
      id: idx + 1,
      name: `Agent #${idx + 1}`,
      avatar: <Icon name="avatar" size={10} />
    }));
    const { container } = render(
      <Followers
        agents={agents}
        isFollowed
        onRemove={jest.fn()}
        onAdd={jest.fn()}
        onFollowToggle={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot("10 followers");
  });

  it("should match the snapshot for 25 followers", () => {
    const agents = Array.from({ length: 25 }, (_, idx) => ({
      id: idx + 1,
      name: `Agent #${idx + 1}`,
      avatar: <Icon name="avatar" size={10} />
    }));
    const { container } = render(
      <Followers
        agents={agents}
        onRemove={jest.fn()}
        onAdd={jest.fn()}
        onFollowToggle={jest.fn()}
      />
    );

    expect(container.firstChild).toMatchSnapshot("25 followers");
  });

  it("should change following state button text", () => {
    const handleFollowToggle = jest.fn();
    let { getByText } = render(
      <Followers
        agents={[
          {
            id: 1,
            name: "Artem Berdyshev",
            avatar: <Icon name="avatar" size={10} />
          }
        ]}
        onRemove={jest.fn()}
        onAdd={jest.fn()}
        onFollowToggle={handleFollowToggle}
      />
    );

    expect(getByText("Follow")).toBeInTheDocument();

    fireEvent.click(getByText("Follow"));
    expect(handleFollowToggle).toBeCalled();

    ({ getByText } = render(
      <Followers
        agents={[
          {
            id: 1,
            name: "Artem Berdyshev",
            avatar: <Icon name="avatar" size={10} />
          }
        ]}
        isFollowed
        onRemove={jest.fn()}
        onAdd={jest.fn()}
        onFollowToggle={handleFollowToggle}
      />
    ));
    expect(getByText("Following")).toBeInTheDocument();
  });

  it("should invoke onAdd event handler on click", () => {
    const handleAdd = jest.fn();
    const { getByText } = render(
      <Followers
        agents={[
          {
            id: 1,
            name: "Artem Berdyshev",
            avatar: <Icon name="avatar" size={10} />
          }
        ]}
        onRemove={jest.fn()}
        onAdd={handleAdd}
        onFollowToggle={jest.fn()}
      />
    );

    expect(getByText("Add")).toBeInTheDocument();

    fireEvent.click(getByText("Add").parentElement);
    expect(handleAdd).toBeCalled();
  });
});
