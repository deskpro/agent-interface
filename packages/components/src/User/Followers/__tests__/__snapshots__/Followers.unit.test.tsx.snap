// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Followers /> should match the snapshot for 10 followers: 10 followers 1`] = `
<div
  class="dp-Followers"
>
  <div
    class="dp-SectionControls"
  >
    <div
      class="dp-SectionTitle"
    >
      <div
        class="dp-TitleControls"
      >
        Followers 
        <span
          class="dp-qt"
        >
          (
          10
          )
        </span>
      </div>
      <button
        class="dp-Button Button--link Button--primary dp-StatusTimes dp-FollowersStatus is-active"
        type="button"
      >
        Following
        <span
          class="dp-Icon Icon--following Icon--primary"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
      </button>
    </div>
  </div>
  <div
    class="dp-FollowerList Followers--Medium"
  >
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #1
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #2
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #3
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #4
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #5
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #6
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #7
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #8
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #9
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #10
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersBtn"
    >
      <button
        class="dp-Button Button--small Button--link Button--primary"
        type="button"
      >
        <span
          class="dp-Icon Icon--add-follower Icon--primary"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span>
          Add
        </span>
      </button>
    </span>
  </div>
</div>
`;

exports[`<Followers /> should match the snapshot for 25 followers: 25 followers 1`] = `
<div
  class="dp-Followers"
>
  <div
    class="dp-SectionControls"
  >
    <div
      class="dp-SectionTitle"
    >
      <div
        class="dp-TitleControls"
      >
        Followers 
        <span
          class="dp-qt"
        >
          (
          25
          )
        </span>
      </div>
      <button
        class="dp-Button Button--link Button--primary dp-StatusTimes dp-FollowersStatus"
        type="button"
      >
        Follow
      </button>
    </div>
  </div>
  <div
    class="dp-FollowerList Followers--Large"
  >
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #1
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #2
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #3
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #4
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #5
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #6
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #7
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #8
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #9
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #10
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #11
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #12
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #13
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #14
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #15
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #16
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #17
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #18
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #19
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #20
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #21
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #22
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #23
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #24
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Agent #25
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersBtn"
    >
      <button
        class="dp-Button Button--small Button--link Button--primary"
        type="button"
      >
        <span
          class="dp-Icon Icon--add-follower Icon--primary"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span>
          Add
        </span>
      </button>
    </span>
  </div>
</div>
`;

exports[`<Followers /> should match the snapshot for single follower: Single follower 1`] = `
<div
  class="dp-Followers"
>
  <div
    class="dp-SectionControls"
  >
    <div
      class="dp-SectionTitle"
    >
      <div
        class="dp-TitleControls"
      >
        Followers 
        <span
          class="dp-qt"
        >
          (
          1
          )
        </span>
      </div>
      <button
        class="dp-Button Button--link Button--primary dp-StatusTimes dp-FollowersStatus"
        type="button"
      >
        Follow
      </button>
    </div>
  </div>
  <div
    class="dp-FollowerList"
  >
    <span
      class="dp-FollowersItem"
    >
      <span
        class="dp-LogoName"
      >
        <span
          class="dp-Icon Icon--avatar"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span
          class="dp-UserName"
        >
          Artem Berdyshev
        </span>
      </span>
      <span
        class="dp-Icon Icon--close"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="8"
          width="8"
        />
      </span>
    </span>
    <span
      class="dp-FollowersBtn"
    >
      <button
        class="dp-Button Button--small Button--link Button--primary"
        type="button"
      >
        <span
          class="dp-Icon Icon--add-follower Icon--primary"
          role="button"
          tabindex="-1"
        >
          <icon-mock
            height="10"
            width="10"
          />
        </span>
        <span>
          Add
        </span>
      </button>
    </span>
  </div>
</div>
`;
